File: src/ProjectR.Scripting/content/scripts/spells/lua/IcicleBarrage.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Icicle Barrage"
end

function GetDescription()
    return "Summon icicles to impale your enemies.\nPierces MR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .3
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ICE }
end

function SpellEffect()
  local damage = (2.5 * aMD * (aICE/100) - .5 * dMR) * (100/dICE)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/IcicleBarrage.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class IcicleBarrage : SpellScriptBase
    {
        public override string Name { get { return "Icicle Barrage"; } }
        public override string Description { get { return "Summon icicles to impale your enemies.\nPierces MR."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ICE, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 30; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (2.5 * cMD * (cICE / 100) - .5 * tMR) * (100 / tICE);
            DealDamage(damage);
        }
    }
}